swap/tests/testutils/mod.rs

```diff
@@ -118,20 +122,27 @@ impl TestContext {
             .await
             .unwrap();
 
-        tokio::spawn(async move { event_loop.run().await });
+        let join_handle = tokio::spawn(async move { event_loop.run().await });
 
-        swap
+        (swap, BobEventLoopJoinHandle(join_handle))
     }
 
-    pub async fn recover_alice_from_db(&mut self) -> alice::Swap {
-        let (swap, mut event_loop) = self.alice_params.builder().await.build().await.unwrap();
+    pub async fn recover_alice_from_db(
+        &mut self,
+        join_handle: AliceEventLoopJoinHandle,
+    ) -> alice::Swap {
+        join_handle.0.abort();
+
+        let (swap, mut event_loop) = self.alice_params.builder().build().await.unwrap();
 
         tokio::spawn(async move { event_loop.run().await });
 
         swap
     }
 
-    pub async fn recover_bob_from_db(&mut self) -> bob::Swap {
+    pub async fn recover_bob_from_db(&mut self, join_handle: BobEventLoopJoinHandle) -> bob::Swap {
```

Comment:
```suggestion
    pub async fn stop_and_resume_bob_from_db(&mut self, join_handle: BobEventLoopJoinHandle) -> bob::Swap {
```
See previous comment for explanation.